wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/drivedistanceoffboard/subsystems/DriveSubsystem.java

```diff
@@ -114,4 +117,72 @@ public void resetEncoders() {
   public void setMaxOutput(double maxOutput) {
     m_drive.setMaxOutput(maxOutput);
   }
+
+  /**
+   * Creates a command to drive forward a specified distance using a motion profile.
+   *
+   * @param distance The distance to drive forward.
+   * @return A command.
+   */
+  public Command profiledDriveDistance(double distance) {
+    var profile =
+        new TrapezoidProfile(
+            new TrapezoidProfile.Constraints(
+                DriveConstants.kMaxSpeedMetersPerSecond,
+                DriveConstants.kMaxAccelerationMetersPerSecondSquared));
+    var timer = new Timer();
+    return startRun(
+            () -> {
+              // Restart timer so profile setpoints start at the beginning
+              timer.restart();
+              resetEncoders();
+            },
+            () -> {
+              // Current state never changes, so we need to use a timer to get the setpoints we need
+              // to be at
+              var setpoint = profile.calculate(timer.get(), new State(), new State(3, 0));
+              setDriveStates(setpoint, setpoint);
+            })
+        .until(() -> profile.isFinished(0));
+  }
+
+  private double m_initialLeftDistance;
+  private double m_initialRightDistance;
+
+  /**
+   * Creates a command to drive forward a specified distance using a motion profile without
+   * resetting the encoders.
+   *
+   * @param distance The distance to drive forward.
+   * @return A command.
+   */
+  public Command dynamicProfiledDriveDistance(double distance) {
+    var profile =
+        new TrapezoidProfile(
+            new TrapezoidProfile.Constraints(
+                DriveConstants.kMaxSpeedMetersPerSecond,
+                DriveConstants.kMaxAccelerationMetersPerSecondSquared));
+    return startRun(
+            () -> {
+              // Store distance so we know the target distance for each encoder
+              m_initialLeftDistance = getLeftEncoderDistance();
+              m_initialRightDistance = getRightEncoderDistance();
+            },
+            () -> {
+              // Current state matches the actual state of the bot, so the profile needs to look 20
```

Comment:
You actually need both the current and future setpoint. Feedback uses the current setpoint, and feedforward uses the current and future setpoint.

Reply:
These examples show usage with smart motor controllers, so all they do is send the setpoints to a "smart motor controller." Implementing PID + FF here is out of scope for this PR.

Reply:
My point is passing the future setpoint here is incorrect.

Reply:
What should I do then? `SetDriveStates` passes the position setpoint to the motor controller, and there's a SimpleMotorFeedForward that's given the velocity setpoint.